File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
    public int solution(int[] elements) {

        Set <Integer> arr = new HashSet<>();
    for(int k =1; k<=elements.length;k++){
        for(int i=0;i<elements.length;i++){
            int sum =0;
            for(int j=i;j<i+k;j++){
                sum+=elements[j%elements.length]; //나눈이유는 0~5까지범위만!!
            }
            arr.add(sum);
        }
    }
        return arr.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    n = len(elements)
    ans = set()
    elements = elements*2
    for i in range(1, n+1):
        for j in range(n):
            s = sum(elements[i:j+i])
            ans.add(s)
            
            
    return len(ans)